crypto/secp256k1eth/secp256k1eth.go

```diff
@@ -0,0 +1,179 @@
+package secp256k1eth
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"crypto/subtle"
+	"fmt"
+	"io"
+	"math/big"
+
+	secp256k1 "github.com/decred/dcrd/dcrec/secp256k1/v4"
+	ethcrypto "github.com/ethereum/go-ethereum/crypto"
```

Comment:
Can we avoid importing the package by default? How about placing it under the build flag `secp256k1eth`?

Reply:
@melekes yes, but let's make sure this lands on `main` first